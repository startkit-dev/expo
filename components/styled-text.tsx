import { Text, type TextProps } from "./themed"

export function MonoText(props: TextProps) {
  return <Text {...props} style={[props.style, { fontFamily: "SpaceMono" }]} />
}
