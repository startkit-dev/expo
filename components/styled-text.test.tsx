import { create } from "react-test-renderer"

import { MonoText } from "./styled-text"

describe("<MonoText />", () => {
  it(`renders correctly`, () => {
    const tree = create(<MonoText>Snapshot test!</MonoText>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
