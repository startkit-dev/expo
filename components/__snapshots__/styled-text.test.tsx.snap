// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonoText /> renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      [
        undefined,
        {
          "fontFamily": "SpaceMono",
        },
      ],
    ]
  }
>
  Snapshot test!
</Text>
`;
